### DISCUSSION TITLE
Two depth buffer in a single pass ?

### DISCUSSION THREAD

**Discussion by jay19240** [gpuweb/gpuweb#4657] [Q&A] [answered]
Hello,
It is possible to have two depth buffers in a single pass ? I want a depth buffer with front-faces and another with back-faces.
Or in other words, first depth buffer with cw and second with ccw.

**Comment by Kangz** [comment, marked as answer]
It isn't possible with depth buffers, there can only be a single one attached.

You could do something like use a depth buffer with depthwrite and depthCompare <= and a R32Float attachment with `GPUBlendOperation` max, but that's kind of a hack and might not work depending on what you want to do with the data afterwards.

**Reply by jay19240** [reply]
Well, so my only non-tricky solution is to create another pipeline with just change primitive culling, right ?

**Reply by Kangz** [reply]
And render all objects again in a new render pass yes.
